activity/scraper.py

```diff
@@ -0,0 +1,95 @@
+import requests
+import json
+import datetime
+from dateutil import parser, relativedelta
+
+# URL to grab all issues from
+issuesurl = "https://coala.github.io/gh-board/issues.json"  # Ignore KeywordBear
+
+# Count of months/weeks/days to be displayed in graph
+month_count = 12
+week_count = 4
+day_count = 7
+today = datetime.datetime.today()
+
+# Calculate the difference in months ignoring partially complete months.
+def diff_month(d1, d2):
+	return (d1.year - d2.year) * 12 + d1.month - d2.month
+
+# Calculate difference in weeks ignoring partially complete weeks.
+def diff_week(d1, d2):
+	monday1 = (d1 - datetime.timedelta(days=d1.weekday()))
+	monday2 = (d2 - datetime.timedelta(days=d2.weekday()))
+	return (monday1 - monday2).days / 7
+
+# Calculate difference in days(ignores and hours difference(since both are dates))
+def diff_days(d1, d2):
+	return (d1-d2).days
+
+# Get labels for each option
+month_labels = []
```

Comment:
you want the python std [`calendar`](https://docs.python.org/3/library/calendar.html) library, which is unfortunately not well known (and will be necessary if we add i18n).

Also move 80% of the remaining code into a class, even if it isnt well designed, because a class will force you to do some basic design, and then the indent levels will be correct , and it is easier to shuffle code around after that has been done.